include/zephyr/net/net_context.h

```diff
@@ -419,6 +419,11 @@ __net_socket struct net_context {
 		};
 	};
 
+#if defined(CONFIG_NET_IPV4)
+	/* Controls whether IPv4 multicast packets are looped back to the sender. */
+	bool ipv4_mcast_loop;
```

Comment:
As I mentioned in IPv6 PR for this feature, we can share this variable between IPv6 and IPv4. So whichever PR is merged 2nd, it needs to change to use union here.